sdks/java/io/tika/src/test/java/org/apache/beam/sdk/io/tika/TikaIOTest.java

```diff
@@ -33,231 +35,148 @@
-    final String[] args = new String[]{"--input=/input/tika.pdf"};
-    TikaIO.Read read = TikaIO.read().withOptions(createOptions(args));
-
-    DisplayData displayData = DisplayData.from(read);
-
-    assertThat(displayData, hasDisplayItem("filePattern", "/input/tika.pdf"));
-    assertThat(displayData, hasDisplayItem("parseMode", "asynchronous"));
-    assertThat(displayData, hasDisplayItem("queuePollTime", "50"));
-    assertThat(displayData, hasDisplayItem("queueMaxPollTime", "3000"));
-    assertEquals(4, displayData.items().size());
-  }
-  @Test
   public void testReadDisplayDataWithCustomOptions() {
     final String[] args = new String[]{"--input=/input/tika.pdf",
                                        "--tikaConfigPath=/tikaConfigPath",
-                                       "--queuePollTime=10",
-                                       "--queueMaxPollTime=1000",
-                                       "--contentTypeHint=application/pdf",
-                                       "--readOutputMetadata=true"};
-    TikaIO.Read read = TikaIO.read().withOptions(createOptions(args));
+                                       "--contentTypeHint=application/pdf"};
+    TikaOptions options = PipelineOptionsFactory.fromArgs(args)
+        .withValidation().as(TikaOptions.class);
+    TikaIO.ParseAll read = TikaIO.parseAll()
+        .withTikaConfigPath(options.getTikaConfigPath())
+        .withContentTypeHint(options.getContentTypeHint());
 
     DisplayData displayData = DisplayData.from(read);
 
-    assertThat(displayData, hasDisplayItem("filePattern", "/input/tika.pdf"));
     assertThat(displayData, hasDisplayItem("tikaConfigPath", "/tikaConfigPath"));
-    assertThat(displayData, hasDisplayItem("parseMode", "asynchronous"));
-    assertThat(displayData, hasDisplayItem("queuePollTime", "10"));
-    assertThat(displayData, hasDisplayItem("queueMaxPollTime", "1000"));
     assertThat(displayData, hasDisplayItem("inputMetadata",
-        "[Content-Type=application/pdf]"));
-    assertThat(displayData, hasDisplayItem("readOutputMetadata", "true"));
-    assertEquals(7, displayData.items().size());
-  }
-
-  private static TikaOptions createOptions(String[] args) {
-    return PipelineOptionsFactory.fromArgs(args)
-        .withValidation().as(TikaOptions.class);
+        "Content-Type=application/pdf"));
+    assertEquals(2, displayData.items().size());
   }
 
-  static class FilterMetadataFn extends DoFn<String, String> {
+  static class FilterMetadataFn extends DoFn<ParseResult, ParseResult> {
     private static final long serialVersionUID = 6338014219600516621L;
 
     @ProcessElement
     public void processElement(ProcessContext c) {
-      String word = c.element();
-      if (word.contains("=") && !word.startsWith("Author")) {
-        return;
+      ParseResult result = c.element();
+      Metadata m = new Metadata();
+      if (result.getFileLocation().endsWith("apache-beam-tika1.odt")) {
```

Comment:
Not sure I'm following what this code is trying to do?

Reply:
I manually created that file and added that specific piece of metadata (Author), but the metadata usually contains some other pieces of info which will also vary between test files so I wanted to test that the metadata I added myself was captured

Reply:
Please add a comment about this